diff --git a/src/components/Toast.tsx b/src/components/Toast.tsx
index c598db60ff3d374db094493fb3a90ae5c4d59077..463219603128d5af80fa842c32b33423efdbbc50 100644
--- a/src/components/Toast.tsx
+++ b/src/components/Toast.tsx
@@ -130,7 +130,7 @@ export const Toast: FC<Props> = ({
           kbHeight -
           insets.bottom -
           (extraInsets?.bottom ?? 0) -
-          24
+          0
         : startingY;
 
       offsetY.value = withSpring(val, {
@@ -241,6 +241,7 @@ export const Toast: FC<Props> = ({
           endPause();
         }}
         onPress={onPress}
+        pointerEvents="box-none"
         style={[
           {
             backgroundColor: !toast.customToast
@@ -267,6 +268,7 @@ export const Toast: FC<Props> = ({
               updateHeight(toast.id, event.nativeEvent.layout.height)
             }
             key={toast.id}
+            pointerEvents="box-none"
           >
             {toast.customToast({
               ...toast,
