diff --git a/android/src/main/java/com/lodev09/truesheet/TrueSheetViewController.kt b/android/src/main/java/com/lodev09/truesheet/TrueSheetViewController.kt
index 502d4704e3a08bd0b8bebd8b2ea2d26dd255d043..e53ba7e2de07614e4c6e65c785efa242cc2a9c97 100644
--- a/android/src/main/java/com/lodev09/truesheet/TrueSheetViewController.kt
+++ b/android/src/main/java/com/lodev09/truesheet/TrueSheetViewController.kt
@@ -74,7 +74,7 @@ class TrueSheetViewController(private val reactContext: ThemedReactContext) :
     private const val DEFAULT_CORNER_RADIUS = 16 // dp
 
     // Animation durations from res/anim/true_sheet_slide_in.xml and true_sheet_slide_out.xml
-    private const val PRESENT_ANIMATION_DURATION = 250L
+    private const val PRESENT_ANIMATION_DURATION = 150L
     private const val DISMISS_ANIMATION_DURATION = 150L
     private const val TRANSLATE_ANIMATION_DURATION = 100L
   }
@@ -688,6 +688,8 @@ class TrueSheetViewController(private val reactContext: ThemedReactContext) :
       paint.color = backgroundColor
     }
     bottomSheet.clipToOutline = true
+    // Removes shadow applied to sheet.
+    bottomSheet.setOutlineProvider(null)
   }
 
   fun setupDimmedBackground(detentIndex: Int) {
diff --git a/android/src/main/java/com/lodev09/truesheet/core/TrueSheetDimView.kt b/android/src/main/java/com/lodev09/truesheet/core/TrueSheetDimView.kt
index 0311c29aaeec70d2491d99e5fb69f4796b30200f..762ab920f7ab66d0435c8cea35d81177774a15a1 100644
--- a/android/src/main/java/com/lodev09/truesheet/core/TrueSheetDimView.kt
+++ b/android/src/main/java/com/lodev09/truesheet/core/TrueSheetDimView.kt
@@ -29,7 +29,8 @@ class TrueSheetDimView(private val reactContext: ThemedReactContext) : View(reac
 
   fun attach(view: ViewGroup? = null, cornerRadius: Float = 0f) {
     if (parent != null) return
-    targetView = view ?: reactContext.currentActivity?.window?.decorView as? ViewGroup
+    // Increase dimness of backdrop by overlaying another one ontop of it.
+    targetView = reactContext.currentActivity?.window?.decorView as? ViewGroup
 
     if (cornerRadius > 0f) {
       outlineProvider = object : ViewOutlineProvider() {
diff --git a/android/src/main/res/anim/true_sheet_slide_in.xml b/android/src/main/res/anim/true_sheet_slide_in.xml
index e5d2bcdaa0c7ca52702208248f403e1189e4e02b..16bd3db3a42457cfb3da1a42ecfd39be9cc3f0a2 100644
--- a/android/src/main/res/anim/true_sheet_slide_in.xml
+++ b/android/src/main/res/anim/true_sheet_slide_in.xml
@@ -1,13 +1,13 @@
 <?xml version="1.0" encoding="utf-8"?>
 <set xmlns:android="http://schemas.android.com/apk/res/android">
     <translate
-        android:duration="250"
+        android:duration="150"
         android:fromYDelta="20%p"
         android:toYDelta="0"
-        android:interpolator="@android:interpolator/decelerate_cubic" />
+        android:interpolator="@android:interpolator/fast_out_slow_in" />
     <alpha
         android:duration="150"
         android:fromAlpha="0.0"
         android:toAlpha="1.0"
-        android:interpolator="@android:interpolator/decelerate_cubic" />
+        android:interpolator="@android:interpolator/fast_out_slow_in" />
 </set>
