diff --git a/android/src/main/java/com/lodev09/truesheet/core/RootSheetView.kt b/android/src/main/java/com/lodev09/truesheet/core/RootSheetView.kt
index e1dcebf2ef7241bcdc5a616c30219428b643caba..915ee4a68e7e96841df116d31005eac03af19398 100644
--- a/android/src/main/java/com/lodev09/truesheet/core/RootSheetView.kt
+++ b/android/src/main/java/com/lodev09/truesheet/core/RootSheetView.kt
@@ -60,7 +60,7 @@ class RootSheetView(private val context: Context?) :
 
   override fun onInterceptTouchEvent(event: MotionEvent): Boolean {
     eventDispatcher?.let { eventDispatcher ->
-      jSTouchDispatcher.handleTouchEvent(event, eventDispatcher, reactContext)
+      jSTouchDispatcher.handleTouchEvent(event, eventDispatcher)
       jSPointerDispatcher?.handleMotionEvent(event, eventDispatcher, true)
     }
     return super.onInterceptTouchEvent(event)
@@ -69,7 +69,7 @@ class RootSheetView(private val context: Context?) :
   @SuppressLint("ClickableViewAccessibility")
   override fun onTouchEvent(event: MotionEvent): Boolean {
     eventDispatcher?.let { eventDispatcher ->
-      jSTouchDispatcher.handleTouchEvent(event, eventDispatcher, reactContext)
+      jSTouchDispatcher.handleTouchEvent(event, eventDispatcher)
       jSPointerDispatcher?.handleMotionEvent(event, eventDispatcher, false)
     }
     super.onTouchEvent(event)
