diff --git a/android/src/main/java/com/lodev09/truesheet/core/TrueSheetAnimator.kt b/android/src/main/java/com/lodev09/truesheet/core/TrueSheetAnimator.kt
index 9bdd76d905b56ff494882e668145e80c1036e78e..7767c4fbac5165bd5f09df1cbef00b640c882c22 100644
--- a/android/src/main/java/com/lodev09/truesheet/core/TrueSheetAnimator.kt
+++ b/android/src/main/java/com/lodev09/truesheet/core/TrueSheetAnimator.kt
@@ -24,7 +24,7 @@ interface TrueSheetAnimatorProvider {
 class TrueSheetAnimator(private val provider: TrueSheetAnimatorProvider) {
 
   companion object {
-    const val PRESENT_DURATION = 300L
+    const val PRESENT_DURATION = 200L
     const val DISMISS_DURATION = 200L
   }
 
diff --git a/android/src/main/java/com/lodev09/truesheet/core/TrueSheetDialogFragment.kt b/android/src/main/java/com/lodev09/truesheet/core/TrueSheetDialogFragment.kt
index 927f6f5a24a2c4329f0d73165a88bf6daae1c105..27442430573030857a93358ae1e91006b0eb276d 100644
--- a/android/src/main/java/com/lodev09/truesheet/core/TrueSheetDialogFragment.kt
+++ b/android/src/main/java/com/lodev09/truesheet/core/TrueSheetDialogFragment.kt
@@ -234,6 +234,8 @@ class TrueSheetDialogFragment : BottomSheetDialogFragment() {
       paint.color = backgroundColor
     }
     bottomSheet.clipToOutline = true
+    // Removes shadow applied to sheet.
+    bottomSheet.setOutlineProvider(null)
   }
 
   fun setupGrabber() {
diff --git a/android/src/main/java/com/lodev09/truesheet/core/TrueSheetDimView.kt b/android/src/main/java/com/lodev09/truesheet/core/TrueSheetDimView.kt
index 5c7a97afc7bb7715aa9ef7cf8501581d7852d61d..463205b2c70cfbdfe43d5e9cf11253803781c286 100644
--- a/android/src/main/java/com/lodev09/truesheet/core/TrueSheetDimView.kt
+++ b/android/src/main/java/com/lodev09/truesheet/core/TrueSheetDimView.kt
@@ -29,7 +29,8 @@ class TrueSheetDimView(private val reactContext: ThemedReactContext) : View(reac
 
   fun attach(view: ViewGroup? = null, cornerRadius: Float = 0f) {
     if (parent != null) return
-    targetView = view ?: reactContext.currentActivity?.window?.decorView as? ViewGroup
+    // Increase dimness of backdrop by overlaying another one ontop of it.
+    targetView = reactContext.currentActivity?.window?.decorView as? ViewGroup
 
     if (cornerRadius > 0f) {
       outlineProvider = object : ViewOutlineProvider() {
