diff --git a/android/src/main/java/expo/modules/adapters/react/permissions/PermissionsService.kt b/android/src/main/java/expo/modules/adapters/react/permissions/PermissionsService.kt
index ae0a81157c193e45c7f22e27ecd66e598535bdc1..754521433cceb1bb6cd0e16f8577fa67583120ec 100644
--- a/android/src/main/java/expo/modules/adapters/react/permissions/PermissionsService.kt
+++ b/android/src/main/java/expo/modules/adapters/react/permissions/PermissionsService.kt
@@ -163,7 +163,7 @@ open class PermissionsService(val context: Context) : InternalModule, Permission
   override fun isPermissionPresentInManifest(permission: String): Boolean {
     try {
       context.packageManager.getPackageInfo(context.packageName, PackageManager.GET_PERMISSIONS)?.run {
-        return requestedPermissions.contains(permission)
+        return requestedPermissions!!.contains(permission)
       }
       return false
     } catch (e: PackageManager.NameNotFoundException) {
