diff --git a/android/src/main/cpp/worklets/WorkletJSCallInvoker.cpp b/android/src/main/cpp/worklets/WorkletJSCallInvoker.cpp
index a2d707ce76ceb35456eb075d06473ac98ad8b5eb..452b8b97b78116d6f93ebba079b931cb831bd2fa 100644
--- a/android/src/main/cpp/worklets/WorkletJSCallInvoker.cpp
+++ b/android/src/main/cpp/worklets/WorkletJSCallInvoker.cpp
@@ -24,7 +24,8 @@ namespace expo {
       return;
     }
 
-    workletRuntime->executeSync([func = std::move(func)](jsi::Runtime &rt) -> jsi::Value {
+    // Temporary Fix: https://github.com/expo/expo/issues/49225
+    workletRuntime->runSync([func = std::move(func)](jsi::Runtime &rt) -> jsi::Value {
       func(rt);
       return jsi::Value::undefined();
     });
diff --git a/ios/WorkletsAdapter/ExpoWorkletsBridgeProvider.mm b/ios/WorkletsAdapter/ExpoWorkletsBridgeProvider.mm
index 126545fae12dc3af71e0ea382b976ef430e62d17..93e56f2e87b369626f8ebe1f0135d5992fee0477 100644
--- a/ios/WorkletsAdapter/ExpoWorkletsBridgeProvider.mm
+++ b/ios/WorkletsAdapter/ExpoWorkletsBridgeProvider.mm
@@ -233,7 +233,8 @@ static jsi::Value callWorklet(jsi::Runtime &rt, std::shared_ptr<worklets::Serial
     return;
   }
 
-  workletRuntime->executeSync([worklet, arguments](jsi::Runtime &rt) -> jsi::Value {
+  // Temporary Fix: https://github.com/expo/expo/issues/49225
+  workletRuntime->runSync([worklet, arguments](jsi::Runtime &rt) -> jsi::Value {
     return callWorklet(rt, worklet, arguments);
   });
 }
