diff --git a/build/hooks.js b/build/hooks.js
index 565bb0187e6fc858b2cc4fefb7f6c5e1ffe73a21..2f095c7a07876c51240ed44a6b9e799712c65a24 100644
--- a/build/hooks.js
+++ b/build/hooks.js
@@ -96,6 +96,9 @@ function useLocalSearchParams() {
                 }),
             ];
         }
+        else if (typeof value === 'undefined' || value === null) {
+            return [key, value];
+        }
         else {
             try {
                 return [key, decodeURIComponent(value)];
