diff --git a/android/build.gradle b/android/build.gradle
index bbca48aed5fc56d5073019fa50ffaf7c9bf5ae3c..f4203320ad68311f1215337bc7ef8527fb403b97 100644
--- a/android/build.gradle
+++ b/android/build.gradle
@@ -82,7 +82,8 @@ android {
         externalNativeBuild {
             cmake {
                 arguments "-DANDROID_STL=c++_shared",
-                        "-DRNS_NEW_ARCH_ENABLED=${IS_NEW_ARCHITECTURE_ENABLED}"
+                        "-DRNS_NEW_ARCH_ENABLED=${IS_NEW_ARCHITECTURE_ENABLED}",
+                        "-DANDROID_SUPPORT_FLEXIBLE_PAGE_SIZES=ON"
             }
         }
     }
