diff --git a/android/src/main/java/com/reactnativeandroidwidget/AndroidWidgetModule.java b/android/src/main/java/com/reactnativeandroidwidget/AndroidWidgetModule.java
index 55ba4fd01e0ee8dbee6e1adc687cd0a0d572a050..4b7c3882a5c185e180bfdef021b0564c156ff959 100644
--- a/android/src/main/java/com/reactnativeandroidwidget/AndroidWidgetModule.java
+++ b/android/src/main/java/com/reactnativeandroidwidget/AndroidWidgetModule.java
@@ -2,6 +2,7 @@ package com.reactnativeandroidwidget;
 
 import android.app.Activity;
 import android.appwidget.AppWidgetManager;
+import android.content.Context;
 import android.content.Intent;
 
 import androidx.annotation.NonNull;
@@ -26,6 +27,17 @@ public class AndroidWidgetModule extends com.reactnativeandroidwidget.AndroidWid
         return NAME;
     }
 
+    @ReactMethod
+    public void openApp() {
+        try {
+            Context context = getReactApplicationContext();
+            Intent launchIntent = context.getPackageManager().getLaunchIntentForPackage(context.getPackageName());
+            context.startActivity(launchIntent);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
     @ReactMethod
     public void drawWidget(ReadableMap config, String widgetName) {
         try {
diff --git a/android/src/main/java/com/reactnativeandroidwidget/oss/HeadlessJsTaskWorker.java b/android/src/main/java/com/reactnativeandroidwidget/oss/HeadlessJsTaskWorker.java
index 813eac4bcf19f84a1939b049efc2d36cae1eb94b..ec18fe0844dac6967a12a19dfcb9a3aad9c4e832 100644
--- a/android/src/main/java/com/reactnativeandroidwidget/oss/HeadlessJsTaskWorker.java
+++ b/android/src/main/java/com/reactnativeandroidwidget/oss/HeadlessJsTaskWorker.java
@@ -33,9 +33,8 @@ import androidx.work.ListenableWorker;
 import androidx.work.WorkerParameters;
 
 import com.facebook.react.ReactApplication;
+import com.facebook.react.ReactHost;
 import com.facebook.react.ReactInstanceEventListener;
-import com.facebook.react.ReactInstanceManager;
-import com.facebook.react.ReactNativeHost;
 import com.facebook.react.bridge.ReactContext;
 import com.facebook.react.bridge.UiThreadUtil;
 import com.facebook.react.jstasks.HeadlessJsTaskConfig;
@@ -72,16 +71,30 @@ public abstract class HeadlessJsTaskWorker extends ListenableWorker implements H
     protected abstract HeadlessJsTaskConfig getTaskConfig(Data data);
 
     protected void startTask(final HeadlessJsTaskConfig taskConfig) {
-        final ReactInstanceManager reactInstanceManager = this.getReactNativeHost().getReactInstanceManager();
-        ReactContext reactContext = reactInstanceManager.getCurrentReactContext();
+        ReactHost reactHost = this.getReactHost();
+        ReactContext reactContext = reactHost.getCurrentReactContext();
         if (reactContext == null) {
-            reactInstanceManager.addReactInstanceEventListener(new ReactInstanceEventListener() {
-                public void onReactContextInitialized(ReactContext reactContext) {
-                    HeadlessJsTaskWorker.this.invokeStartTask(reactContext, taskConfig);
-                    reactInstanceManager.removeReactInstanceEventListener(this);
-                }
-            });
-            reactInstanceManager.createReactContextInBackground();
+            //! For some weird reason, `startWork` fires even after we close the app
+            //! (when the work should be completed already), causing `startTask` to fire.
+            //! Since the React Native instance doesn't exist since the app has been closed,
+            //! we enter this block and re-initialize the React Native instance, causing all
+            //! the on-start-up logic to fire (RNTP service setup, widget setup).
+            //!
+            //! THIS RESULTS IN THE APP TO ESSENTIALLY INFINITELY SPIN BACK UP WHEN WE DON'T
+            //! WANT IT TO (ie: in dev mode, we'll see `Android Bundled` in the console
+            //! AFTER WE CLOSED THE APP, and then the console spammed by `android_cannot_setup_player_in_background`).
+            //!
+            //! This might be caused to also having a `HeadlessJsTaskEventListener` for
+            //! RNTP.
+            //!
+            //! Commenting out the below code seems to fix the issue.
+            // reactHost.addReactInstanceEventListener(new ReactInstanceEventListener() {
+            //     public void onReactContextInitialized(ReactContext reactContext) {
+            //         HeadlessJsTaskWorker.this.invokeStartTask(reactContext, taskConfig);
+            //         reactHost.removeReactInstanceEventListener(this);
+            //     }
+            // });
+            // reactHost.start();
         } else {
             this.invokeStartTask(reactContext, taskConfig);
         }
@@ -94,9 +107,9 @@ public abstract class HeadlessJsTaskWorker extends ListenableWorker implements H
     }
 
     private void cleanUpTask() {
-        if (this.getReactNativeHost().hasInstance()) {
-            ReactInstanceManager reactInstanceManager = this.getReactNativeHost().getReactInstanceManager();
-            ReactContext reactContext = reactInstanceManager.getCurrentReactContext();
+        ReactHost reactNativeHost = this.getReactHost();
+        if (reactNativeHost != null) {
+            ReactContext reactContext = reactNativeHost.getCurrentReactContext();
             if (reactContext != null) {
                 HeadlessJsTaskContext headlessJsTaskContext = HeadlessJsTaskContext.getInstance(reactContext);
                 headlessJsTaskContext.removeTaskEventListener(this);
@@ -125,7 +138,7 @@ public abstract class HeadlessJsTaskWorker extends ListenableWorker implements H
         }
     }
 
-    protected ReactNativeHost getReactNativeHost() {
-        return ((ReactApplication) this.getApplicationContext()).getReactNativeHost();
+    private ReactHost getReactHost() {
+        return ((ReactApplication) this.getApplicationContext()).getReactHost();
     }
 }
diff --git a/android/src/oldarch/AndroidWidgetSpec.java b/android/src/oldarch/AndroidWidgetSpec.java
index 09e70c4412d411e6137bfd6a1aed09e45474c4c1..644f39e87dc4778b3546b53fe875bbe9f1a357cd 100644
--- a/android/src/oldarch/AndroidWidgetSpec.java
+++ b/android/src/oldarch/AndroidWidgetSpec.java
@@ -12,6 +12,8 @@ abstract class AndroidWidgetSpec extends ReactContextBaseJavaModule {
         super(context);
     }
 
+    public abstract void openApp();
+
     public abstract void drawWidget(ReadableMap config, String widgetName);
 
     public abstract void drawWidgetById(ReadableMap config, String widgetName, double widgetId);
diff --git a/src/AndroidWidget.ts b/src/AndroidWidget.ts
index f0d225d2f4e50e6e005cc478fd65f5e35202d8d9..376b833159c3d18414f9879542ee4fd4fb9ac028 100644
--- a/src/AndroidWidget.ts
+++ b/src/AndroidWidget.ts
@@ -10,6 +10,7 @@ const LINKING_ERROR =
 const isTurboModuleEnabled = global.__turboModuleProxy != null;
 
 const noopModule: Spec = {
+  openApp: () => null,
   drawWidget: () => null,
   drawWidgetById: () => null,
   createPreview: () =>
diff --git a/src/NativeAndroidWidget.ts b/src/NativeAndroidWidget.ts
index fabefa3658481fefaf5832e9e3359def47055b21..9bf0b27c350f7cae1322104e8b82ec9deaa2edcd 100644
--- a/src/NativeAndroidWidget.ts
+++ b/src/NativeAndroidWidget.ts
@@ -4,6 +4,7 @@ import type { WidgetPreviewData } from './api/private.types';
 import type { WidgetInfo } from './api/types';
 
 export interface Spec extends TurboModule {
+  openApp(): void;
   drawWidget(config: Object, widgetName: string): void;
   drawWidgetById(config: Object, widgetName: string, widgetId: number): void;
   createPreview(
diff --git a/src/api/open-app.ts b/src/api/open-app.ts
new file mode 100644
index 0000000000000000000000000000000000000000..f007f020282b2795fae08e3a78abc27996015c3c
--- /dev/null
+++ b/src/api/open-app.ts
@@ -0,0 +1,8 @@
+import { AndroidWidget } from '../AndroidWidget';
+
+/**
+ * Open app with `startActivity`.
+ */
+export function openApp(): void {
+  return AndroidWidget.openApp();
+}
diff --git a/src/index.ts b/src/index.ts
index 31ff37c23ee2b9b9bac693842642df8036a7cd5a..33d18a7dc0f91caf5e00aa664d519a5ba7e497a6 100644
--- a/src/index.ts
+++ b/src/index.ts
@@ -1,5 +1,6 @@
 export * from './api/WidgetPreview';
 export * from './api/get-widget-info';
+export * from './api/open-app';
 export * from './api/register-widget-configuration-screen';
 export * from './api/register-widget-task-handler';
 export * from './api/request-widget-update';
