diff --git a/dist/commonjs/index.js b/dist/commonjs/index.js
index 4171a10c0a2e8729292e2f9ede86703b7a879bfd..635b38120777d520f1e89b5f7a6ef36c5b28d519 100644
--- a/dist/commonjs/index.js
+++ b/dist/commonjs/index.js
@@ -19,6 +19,7 @@ function hide(config = {}) {
   } = config;
   return _NativeRNBootSplash.default.hide(fade).then(() => {});
 }
+export const isSystemDarkMode = _NativeRNBootSplash.default.getConstants().darkModeEnabled;
 function isVisible() {
   return _NativeRNBootSplash.default.isVisible();
 }
@@ -154,6 +155,7 @@ function useHideAnimation(config) {
 }
 var _default = exports.default = {
   hide,
+  isSystemDarkMode,
   isVisible,
   useHideAnimation
 };
diff --git a/dist/module/index.js b/dist/module/index.js
index 69bf79e04c1dfeed50e3b6265913d96b2f7bffbb..f7298ac52ba3df719f5666a7214292e35b04ec69 100644
--- a/dist/module/index.js
+++ b/dist/module/index.js
@@ -11,6 +11,7 @@ export function hide(config = {}) {
   } = config;
   return NativeModule.hide(fade).then(() => {});
 }
+export const isSystemDarkMode = NativeModule.getConstants().darkModeEnabled;
 export function isVisible() {
   return NativeModule.isVisible();
 }
@@ -146,6 +147,7 @@ export function useHideAnimation(config) {
 }
 export default {
   hide,
+  isSystemDarkMode,
   isVisible,
   useHideAnimation
 };
diff --git a/dist/typescript/index.d.ts b/dist/typescript/index.d.ts
index 9f59e3730d028bd4553d81a5db3d41ca96c2b4d7..f6f993865451ad5b9c7a97905b3b0ae5abb22723 100644
--- a/dist/typescript/index.d.ts
+++ b/dist/typescript/index.d.ts
@@ -50,10 +50,12 @@ export type UseHideAnimation = {
     brand: BrandProps;
 };
 export declare function hide(config?: Config): Promise<void>;
+export declare const isSystemDarkMode: boolean;
 export declare function isVisible(): boolean;
 export declare function useHideAnimation(config: UseHideAnimationConfig): UseHideAnimation;
 declare const _default: {
     hide: typeof hide;
+    isSystemDarkMode: typeof isSystemDarkMode;
     isVisible: typeof isVisible;
     useHideAnimation: typeof useHideAnimation;
 };
diff --git a/src/index.ts b/src/index.ts
index 1761343ad14e7119bc8ef3e59504f02d569fdcde..8cf11e732ceff31871d5a74df15ed5acaa431857 100644
--- a/src/index.ts
+++ b/src/index.ts
@@ -81,6 +81,8 @@ export function hide(config: Config = {}): Promise<void> {
   return NativeModule.hide(fade).then(() => {});
 }
 
+export const isSystemDarkMode = NativeModule.getConstants().darkModeEnabled;
+
 export function isVisible(): boolean {
   return NativeModule.isVisible();
 }
@@ -278,6 +280,7 @@ export function useHideAnimation(config: UseHideAnimationConfig) {
 
 export default {
   hide,
+  isSystemDarkMode,
   isVisible,
   useHideAnimation,
 };
