diff --git a/dist/common/components/native/RNGHPressable.js b/dist/common/components/native/RNGHPressable.js
new file mode 100644
index 0000000000000000000000000000000000000000..a9ff99027d1756ef814213cf9ca195ebb767fba2
--- /dev/null
+++ b/dist/common/components/native/RNGHPressable.js
@@ -0,0 +1,31 @@
+"use strict";
+
+Object.defineProperty(exports, "__esModule", {
+  value: true
+});
+module.exports = exports.RNGHPressable = void 0;
+var _jsxRuntime = require("react/jsx-runtime");
+var _reactNativeGestureHandler = require("react-native-gesture-handler");
+var _context = require("../../core/context");
+var _native = require("../../core/native");
+var _utils = require("../utils");
+var _useStyle = require("./useStyle");
+const RNGHPressable = exports.RNGHPressable = (0, _utils.copyComponentProperties)(_reactNativeGestureHandler.Pressable, props => {
+  const style = (0, _useStyle.useStyle)(props.className, props, {
+    isDisabled: Boolean(props.disabled)
+  });
+  const uniwindContext = (0, _context.useUniwindContext)();
+  return /* @__PURE__ */(0, _jsxRuntime.jsx)(_reactNativeGestureHandler.Pressable, {
+    ...props,
+    style: state => {
+      if (state.pressed) {
+        return [_native.UniwindStore.getStyles(props.className, props, {
+          isDisabled: Boolean(props.disabled),
+          isPressed: true
+        }, uniwindContext).styles, typeof props.style === "function" ? props.style(state) : props.style];
+      }
+      return [style, typeof props.style === "function" ? props.style(state) : props.style];
+    }
+  });
+});
+module.exports = RNGHPressable;
\ No newline at end of file
diff --git a/dist/common/index.js b/dist/common/index.js
index 63a300fd5dd3ce2f01ec65b65e0ed7aee68c4cda..515df3bd15e6d10cea185970bb0abfbf82445027 100644
--- a/dist/common/index.js
+++ b/dist/common/index.js
@@ -8,8 +8,15 @@ var _exportNames = {
   withUniwind: true,
   useCSSVariable: true,
   useResolveClassNames: true,
-  useUniwind: true
+  useUniwind: true,
+  RNGHPressable: true
 };
+Object.defineProperty(exports, "RNGHPressable", {
+  enumerable: true,
+  get: function () {
+    return _RNGHPressable.default;
+  }
+});
 Object.defineProperty(exports, "Uniwind", {
   enumerable: true,
   get: function () {
@@ -55,3 +62,5 @@ Object.keys(_ScopedTheme).forEach(function (key) {
 var _core = require("./core");
 var _hoc = require("./hoc");
 var _hooks = require("./hooks");
+var _RNGHPressable = _interopRequireDefault(require("./components/native/RNGHPressable"));
+function _interopRequireDefault(e) { return e && e.__esModule ? e : { default: e }; }
\ No newline at end of file
diff --git a/dist/module/components/native/RNGHPressable.d.ts b/dist/module/components/native/RNGHPressable.d.ts
new file mode 100644
index 0000000000000000000000000000000000000000..5db19b2d860a8a5672959ea9536d31f8b6b06b29
--- /dev/null
+++ b/dist/module/components/native/RNGHPressable.d.ts
@@ -0,0 +1,3 @@
+import { PressableProps } from 'react-native-gesture-handler';
+export declare const RNGHPressable: (props: PressableProps) => React.JSX.Element;
+export default RNGHPressable;
diff --git a/dist/module/components/native/RNGHPressable.js b/dist/module/components/native/RNGHPressable.js
new file mode 100644
index 0000000000000000000000000000000000000000..12d7626de7db8b98f5e0dff58c7df26aa459521c
--- /dev/null
+++ b/dist/module/components/native/RNGHPressable.js
@@ -0,0 +1,37 @@
+import { jsx } from "react/jsx-runtime";
+import { Pressable as RNPressable } from "react-native-gesture-handler";
+import { useUniwindContext } from "../../core/context.js";
+import { UniwindStore } from "../../core/native/index.js";
+import { copyComponentProperties } from "../utils.js";
+import { useStyle } from "./useStyle.js";
+export const RNGHPressable = copyComponentProperties(RNPressable, (props) => {
+  const style = useStyle(
+    props.className,
+    props,
+    {
+      isDisabled: Boolean(props.disabled)
+    }
+  );
+  const uniwindContext = useUniwindContext();
+  return /* @__PURE__ */ jsx(
+    RNPressable,
+    {
+      ...props,
+      style: (state) => {
+        if (state.pressed) {
+          return [
+            UniwindStore.getStyles(
+              props.className,
+              props,
+              { isDisabled: Boolean(props.disabled), isPressed: true },
+              uniwindContext
+            ).styles,
+            typeof props.style === "function" ? props.style(state) : props.style
+          ];
+        }
+        return [style, typeof props.style === "function" ? props.style(state) : props.style];
+      }
+    }
+  );
+});
+export default RNGHPressable;
diff --git a/dist/module/index.d.ts b/dist/module/index.d.ts
index c1d990fcb3e4b2ddd411deaa839fc15f39efb971..271ff86304f4674f8f2b2b1356c0621a8c5ef64b 100644
--- a/dist/module/index.d.ts
+++ b/dist/module/index.d.ts
@@ -4,3 +4,5 @@ export { withUniwind } from './hoc';
 export type { ApplyUniwind, ApplyUniwindOptions } from './hoc/types';
 export { useCSSVariable, useResolveClassNames, useUniwind } from './hooks';
 export type { UniwindConfig } from './types';
+import RNGHPressable from './components/native/RNGHPressable';
+export { RNGHPressable };
diff --git a/dist/module/index.js b/dist/module/index.js
index b4fca540efd7bba07af6223f195836b77f2b629e..aeaf35e54bc6d762971cf265dc253106e155f2fa 100644
--- a/dist/module/index.js
+++ b/dist/module/index.js
@@ -2,3 +2,5 @@ export * from "./components/ScopedTheme/index.js";
 export { Uniwind } from "./core/index.js";
 export { withUniwind } from "./hoc/index.js";
 export { useCSSVariable, useResolveClassNames, useUniwind } from "./hooks/index.js";
+import RNGHPressable from "./components/native/RNGHPressable.js";
+export { RNGHPressable };
diff --git a/package.json b/package.json
index 1f2f8d4ea4b2590d3cbe832caf515ecc175113cc..40832bae1f4d28f30cb512516f6dd31c9f25a1c7 100644
--- a/package.json
+++ b/package.json
@@ -86,6 +86,7 @@
     "peerDependencies": {
         "react": ">=19.0.0",
         "react-native": ">=0.81.0",
+        "react-native-gesture-handler": ">=2.28.0",
         "tailwindcss": ">=4"
     },
     "devDependencies": {
diff --git a/src/components/native/RNGHPressable.tsx b/src/components/native/RNGHPressable.tsx
new file mode 100644
index 0000000000000000000000000000000000000000..80d8ba8d98dc9f527b1a79bce319982edcfa0ebd
--- /dev/null
+++ b/src/components/native/RNGHPressable.tsx
@@ -0,0 +1,39 @@
+import { Pressable as RNPressable, PressableProps } from 'react-native-gesture-handler'
+import { useUniwindContext } from '../../core/context'
+import { UniwindStore } from '../../core/native'
+import { copyComponentProperties } from '../utils'
+import { useStyle } from './useStyle'
+
+export const RNGHPressable = copyComponentProperties(RNPressable, (props: PressableProps) => {
+    const style = useStyle(
+        props.className,
+        props,
+        {
+            isDisabled: Boolean(props.disabled),
+        },
+    )
+    const uniwindContext = useUniwindContext()
+
+    return (
+        <RNPressable
+            {...props}
+            style={state => {
+                if (state.pressed) {
+                    return [
+                        UniwindStore.getStyles(
+                            props.className,
+                            props,
+                            { isDisabled: Boolean(props.disabled), isPressed: true },
+                            uniwindContext,
+                        ).styles,
+                        typeof props.style === 'function' ? props.style(state) : props.style,
+                    ]
+                }
+
+                return [style, typeof props.style === 'function' ? props.style(state) : props.style]
+            }}
+        />
+    )
+})
+
+export default RNGHPressable
diff --git a/src/index.ts b/src/index.ts
index daeba02cba25482d84d208dbaa2973453104cc8e..8420ff0ee393dc8e03498042eaba4ee465803f56 100644
--- a/src/index.ts
+++ b/src/index.ts
@@ -4,3 +4,6 @@ export { withUniwind } from './hoc'
 export type { ApplyUniwind, ApplyUniwindOptions } from './hoc/types'
 export { useCSSVariable, useResolveClassNames, useUniwind } from './hooks'
 export type { UniwindConfig } from './types'
+
+import RNGHPressable from './components/native/RNGHPressable'
+export { RNGHPressable }
